### PR TITLE
Add ram test and remove unused header

### PR DESCRIPTION
**Pull request by vnghia**
@mihaimaruseac 
This is the test for `ram_file_block_cache`. I keep almost everything so it should be fine. ( only `status` to `TF_Status` as usual)

### CODE REVIEW

File: tensorflow/c/experimental/filesystem/plugins/gcs/ram_file_block_cache_test.cc [hunk 1]
```diff
@@ -0,0 +1,589 @@
...
+  EXPECT_TRUE(cache4.IsCacheEnabled());
+}
+
+TEST(RamFileBlockCacheTest, ValidateAndUpdateFileSignature) {
+  int calls = 0;
+  auto fetcher = [&calls](const string& filename, size_t offset, size_t n,
+                          char* buffer, size_t* bytes_transferred,
+                          TF_Status* status) {
+    calls++;
+    memset(buffer, 'x', n);
```
Comment (mihaimaruseac) [inline tensorflow/c/experimental/filesystem/plugins/gcs/ram_file_block_cache_test.cc:68]: Why not `memset` with 0? Or why `memset` at all? Shouldn't the caller be responsible for initial data here?

**Review by mihaimaruseac — APPROVED** [review]

File: tensorflow/c/experimental/filesystem/plugins/gcs/ram_file_block_cache_test.cc (hunk 1 above)
Comment (vnvo2409) [reply tensorflow/c/experimental/filesystem/plugins/gcs/ram_file_block_cache_test.cc:68]: No. The caller is responsible for allocating memory for `buffer`. But this function is a callback to fetch the data if the data we need isn't cached yet. So this function has to fill some data to `buffer`

Comment (vnvo2409) [reply tensorflow/c/experimental/filesystem/plugins/gcs/ram_file_block_cache_test.cc:68]: Here: https://github.com/tensorflow/tensorflow/blob/a61a816f86bb35d1d3a58f1c9a236be5e14c55c7/tensorflow/c/experimental/filesystem/plugins/gcs/ram_file_block_cache.h#L98-L99

And here:
https://github.com/tensorflow/tensorflow/blob/a61a816f86bb35d1d3a58f1c9a236be5e14c55c7/tensorflow/c/experimental/filesystem/plugins/gcs/ram_file_block_cache.h#L42-L50

Comment (mihaimaruseac) [reply tensorflow/c/experimental/filesystem/plugins/gcs/ram_file_block_cache_test.cc:68]: Oh, so we are mocking filling in the data. Sounds good.

Comment (vnvo2409) [reply tensorflow/c/experimental/filesystem/plugins/gcs/ram_file_block_cache_test.cc:68]: That's right

**Pull request merged**